### ISSUE TITLE
[Bug] 域名设置好并且部署成功一次后，后面再新增的其他部署不会执行

### ISSUE THREAD

**Issue by hucanye** [usual2970/certimate#287]
**描述问题**
域名设置好并且部署成功一次后，后面再新增的其他部署方式不会执行部署

**复现步骤**
复现该问题的步骤：

1. 域名设置好，部署了一台服务器，点击部署，部署成功
2. 后期发现有新的需求，这个域名证书还要部署其他服务器，于是编辑域名新增其他部署方式
3. 点部署，提示“证书在有效期内且已部署，跳过”，新增的部署方式不会执行

**期望的结果**
应该判断设置的每个部署是否都执行了，而不是证书部署过就不继续执行后面的部署了。
建议点部署的可以自行勾选执行哪些部署。

**截图**
![image](https://github.com/user-attachments/assets/89d4e38a-f9a1-4b45-855f-75b10ddd0e28)

**环境**

仓库版本: [e.g. v0.2.11]

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
To solve the issue of not deploying new configurations when the certificate is still valid, we need to modify the `deploy` function in `internal/domains/deploy.go`. We'll remove the early return when the certificate is valid and already deployed, allowing the function to proceed with deploying new configurations. We'll also update the deployment logic to check each deploy configuration individually.

Here's a summary of the changes:
1. Remove the early return when the certificate is valid and deployed.
2. Move the certificate check after the deployment loop.
3. Update the deployment loop to check and execute each configuration.
4. Update the history recording to reflect the new behavior.

I also created pull request: https://github.com/usual2970/certimate/pull/288

> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
